account-integrations/webauthn/src/SafeECDSAPlugin.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+pragma abicoder v2;
+
+import {BaseAccount} from "account-abstraction/contracts/core/BaseAccount.sol";
+import {IEntryPoint, UserOperation} from "account-abstraction/contracts/interfaces/IEntryPoint.sol";
+import {UserOperation} from "account-abstraction/contracts/interfaces/IEntryPoint.sol";
+
+import {ECDSA} from "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
+
+interface ISafe {
+    function enableModule(address module) external;
+
+    function execTransactionFromModule(
+        address to,
+        uint256 value,
+        bytes memory data,
+        uint8 operation
+    ) external returns (bool success);
+}
+
+contract SafeECDSAPlugin is BaseAccount {
+    using ECDSA for bytes32;
+
+    address public immutable myAddress;
+    address private immutable _owner;
+    address private immutable _entryPoint;
+
+    address internal constant _SENTINEL_MODULES = address(0x1);
+
+    constructor(address entryPointAddress, address ownerAddress) {
+        myAddress = address(this);
+        _owner = ownerAddress;
+        _entryPoint = entryPointAddress;
+    }
+
+    function validateUserOp(
+        UserOperation calldata userOp,
+        bytes32 userOpHash,
+        uint256 missingAccountFunds
+    ) external override returns (uint256 validationData) {
+        address payable safeAddress = payable(userOp.sender);
+        ISafe senderSafe = ISafe(safeAddress);
+
+        if (missingAccountFunds != 0) {
+            senderSafe.execTransactionFromModule(
+                _entryPoint,
+                missingAccountFunds,
+                "",
+                0
+            );
+        }
+
+        validationData = _validateSignature(userOp, userOpHash);
+    }
+
+    function execTransaction(
+        address to,
+        uint256 value,
+        bytes calldata data
+    ) external payable {
+        address payable safeAddress = payable(msg.sender);
+        ISafe safe = ISafe(safeAddress);
+        require(
+            safe.execTransactionFromModule(to, value, data, 0),
+            "tx failed"
+        );
+    }
+
+    function enableMyself() public {
+        ISafe(address(this)).enableModule(myAddress);
```

Comment:
1. What is this enableMyself() function?
It's used as a wrapper around the the Safe's `enableModule` function, which is then called from the `setup` function [here](https://github.com/getwax/wax/blob/38368a72a6c8a77db41825c3791ed79018e32edf/account-integrations/webauthn/test/integration/SafeECDSAPlugin.test.ts#L114). The setup function will then call the `setupModules` function [here](https://github.com/safe-global/safe-contracts/blob/af488f0a649a3be2ea01e0f7a6af1587d0250c2d/contracts/base/ModuleManager.sol#L27-L41). I took the function from the safe integration test and if I'm honest, I'm not sure why they chose to create this wrapper function instead of just call `enableModule`. I noticed other modules  using older safe code (when it was called gnosis safe) inherited from [a contract](https://github.com/safe-global/safe-modules/blob/master/recurring_transfers/contracts/RecurringTransfersModule.sol) called `Module`. I would need to look into that further to figure out what is happening under the hood. This is probably something to come back to and assess whether it's needed for our use case in the future, will add an issue for it. 

3. isn't address(this) the same as myAddress
My assumption would be, since this is getting called from the context of the setup function in Safe.sol, `address(this)` is equal to the address of Safe.sol, and not the plugin. Again, not 100% sure on how everything in Safe fits together yet, so would want to verify this at some point